### PR TITLE
[docs] reorder items for configuration reference page

### PR DESCRIPTION
**Pull request by sarah11918**
## Changes

This reorders the `@docs` items for building the Docs Configuration Reference Page. There is **no content change**, only organizational changes. This should only affect how the docs page is generated (the order of items).

(File diffs may imply otherwise, because of similar text repeated throughout. But there have been no changes to any text. Only moving items around and attempting to get the indentation correct.)

- It reorganizes similar items together in an order that more reflects what people are asking about and looking for.

- It moves `integrations` and `vite` into the top-level category instead of making them standalone headers at the bottom of the page, mixed in with config items that *do* have several sub-options, taking care to:
	- [x]  remove the `@heading` property
	- [x] change the name to lower case

## Testing

No tests, only docs. Not sure why tests are failing.

## Docs

I hope I got all the indentations correct

### CODE REVIEW

File: packages/astro/src/@types/astro.ts [hunk 1]
```diff
@@ -675,48 +745,36 @@ export interface AstroUserConfig {
...
-	 */
-	adapter?: AstroIntegration;
-
-	/**
-	 * @docs
-	 * @name output
-	 * @type {('static' | 'server' | 'hybrid')}
-	 * @default `'static'`
-	 * @see adapter
+	 * @name vite
```
Comment (TheOtterlord) [inline packages/astro/src/@types/astro.ts:748]:
```suggestion
	 * @name vite
	 * @typeraw {ViteUserConfig}
```

File: packages/astro/src/@types/astro.ts [hunk 2]
```diff
@@ -556,104 +546,184 @@ export interface AstroUserConfig {
...
 	 * ```
 	 */
-	compressHTML?: boolean;
+	adapter?: AstroIntegration;
 
 	/**
 	 * @docs
-	 * @name base
-	 * @type {string}
+	 * @name integrations
```
Comment (TheOtterlord) [inline packages/astro/src/@types/astro.ts:595]:
```suggestion
	 * @name integrations
	 * @typeraw {AstroIntegration[]}
```

**Review by TheOtterlord — APPROVED** [review]
Adding types for `integrations` & `vite`, but looks good otherwise

**Review by natemoo-re — APPROVED** [review]
LGTM!

**Pull request merged**